src/cli/commands/test/cache.rs

```diff
@@ -0,0 +1,31 @@
+pub mod cache {
+	use std::{fmt::Debug, fs::read_to_string, path::PathBuf};
+
+	use thiserror::Error;
+
+	use serde::{Deserialize, Serialize};
+	use serde_json;
+
+	#[derive(Error, Debug)]
+	pub enum CacheError {
+		#[error("file `{0}` not found ")]
+		FileNotFound(PathBuf),
```

Comment:
You can probably use the `from` keyword from thiserror
example:
```rust
use thiserror::Error;

#[derive(Error, Debug)]
pub enum DataStoreError {
    #[error("data store disconnected")]
    Disconnect(#[from] io::Error),
    #[error("unknown data store error")]
    Unknown,
}
```
That way you can directly transform a `read_to_string` error to our CacheError type.

This will spare you the `.map_err ` step, as `?` should directly transform your error.

Reply:
https://crates.io/crates/thiserror/1.0.24